### PR TITLE
Build macOS arm64 wheels on macos-14; Build abi3 wheel only; Add typing

### PR DESCRIPTION
**Pull request by laggykiller**
Solves #58 #59 #60

### CODE REVIEW

File: webp_build/builder.py [hunk 1]
```diff
@@ -42,10 +43,10 @@ def get_arch() -> str:
     raise RuntimeError('Unable to determine the compilation target architecture')
 
 
-def install_libwebp(arch: str) -> dict:
+def install_libwebp(arch: str) -> "dict[Any, Any]":
```
Comment (anibali) [inline webp_build/builder.py:46]: I think that the most Python 3.8-friendly way of handling dict and list would be to use typing.Dict and typing.List

File: webp/__init__.py [hunk 2]
```diff
@@ -553,7 +573,13 @@ def _mimwrite_pics(file_path, pics, *, fps=30.0, **kwargs):
         f.write(anim_data.buffer())
 
 
-def mimwrite(file_path, arrs, *, fps=30.0, pilmode=None, **kwargs):
+def mimwrite(
+        file_path: str,
+        arrs: "List[np.ndarray[Any, Any]]",
```
Comment (anibali) [inline webp/__init__.py:578]: Quoting should not be necessary here

**Comment by anibali** [comment]
This is looking really good, thanks for the PR! I just have a few requests based on my preferences with respect to typing.

**Comment by laggykiller** [comment]
Fixed, also I noticed that I missed quite a few of typing, please check again, thanks!

**Comment by anibali** [comment]
Fantastic, thanks for your contribution!

**Pull request merged**